python/ray/tune/trial_runner.py

```diff
@@ -28,6 +29,15 @@ def _naturalize(string):
     return [int(text) if text.isdigit() else text.lower() for text in splits]
 
 
+def _find_newest_ckpt(ckpt_dir):
+    """Returns path to most recently modified checkpoint."""
+    full_paths = [
+        os.path.join(ckpt_dir, fname) for fname in os.listdir(ckpt_dir)
+        if fname.startswith("experiment_state") and fname.endswith(".json")
+    ]
+    return max(full_paths, key=lambda f: os.path.getmtime(f))
```

Comment:
Can we instead go by lexiographic sort? mtime isn't preserved in many situations